Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -52,48 +53,273 @@ public static class CustomEntitySearch
-                                        new Entity
-                                        {
-                                            Name = match.Value,
-                                            MatchIndex = match.Index
-                                        });
+                                    if (!entityMatch.Where(x => (x.Index >= currTextChar - (wordCharArray.Count + leniency)) &&
+                                    (x.Index <= currTextChar - wordCharArray.Count)).Any())
+                                    {
+                                        valuesList.Add(wordFound.ToString());
+                                        offsetList.Add(currTextChar - wordFound.ToString().Length);
+                                        confidenceList.Add(currMismatch + (wordCharArray.Count - currWordChar) / leniency);
+                                        if (minVal < 0 || wordFound.ToString().Length < minVal)
+                                            minVal = wordFound.ToString().Length;
+                                        addWord = true;
+                                    }
+                                    currWordChar = KMPTable[currWordChar];
+                                    wordFound.GetStringBuilder().Clear();
+                                    currMismatch = 0;
                                 }
-                                entitiesFound.Add(word);
+                                else
+                                {
+                                    currWordChar = KMPTable[currWordChar];
+                                    if (currWordChar < 0)
+                                    {
+                                        currWordChar++;
+                                        currTextChar++;
+                                    }
+                                    wordFound.GetStringBuilder().Clear();
+                                    currMismatch = 0;
+                                }
+                            }
+                        }
+                        else
+                        {
+                            string accents = @"[\p{Mn}\p{Sk}\p{Mc}]";
+                            bool wordHalfMark = Regex.Match(wordCharArray[currWordChar].ToString(), accents).Success;
+                            bool textHalfMark = Regex.Match(textCharArray[currTextChar].ToString(), accents).Success;
+
+                            // fuzzy situation?
+                            // accent case adds 0.5
+                            if ((wordHalfMark && textHalfMark) ||
+                                String.Compare(wordCharArray[currWordChar].ToString(), textCharArray[currTextChar].ToString(),
+                                CultureInfo.CurrentCulture, CompareOptions.IgnoreCase | CompareOptions.IgnoreNonSpace) == 0)
+                            {
+                                currMismatch += 0.5;
+                                wordFound.Write(text.ElementAt(currTextChar));
```

Comment:
why mismatch += 0.5 if the strings match?

Reply:
They have different accents or one has an accent and the other does not

Reply:
When we have the accent sensitivity flag, this should become 0 or 1.